### PR TITLE
Embedding binary enhancements

### PR DESCRIPTION
**Pull request by nsosio**
- [x] added configurable port
- [x] added model-path paramether

### CODE REVIEW

File: ebd-all-minilm/models.py [hunk 1]
```diff
@@ -12,10 +12,12 @@ def embeddings(cls, texts):
         return values.tolist()
 
     @classmethod
-    def get_model(cls):
+    def get_model(cls, model=None):
         if cls.model is None:
+            if model is None or not os.path.exists(model):
+                model = os.getenv("MODEL_ID", "all-MiniLM-L6-v2")
```
Comment (casperdcl) [inline ebd-all-minilm/models.py:18]:
```suggestion
    def get_model(cls, model_path):
        if cls.model is None:
```

File: ebd-all-minilm/models.py [hunk 2]
```diff
@@ -12,10 +12,12 @@ def embeddings(cls, texts):
...
             cls.model = SentenceTransformer(
-                os.getenv("MODEL_ID", "all-MiniLM-L6-v2"),
+                model,
```
Comment (casperdcl) [inline ebd-all-minilm/models.py:20]:
```suggestion
                model_path,
```

**Review by casperdcl — CHANGES_REQUESTED** [review]
same approach as `cht-llama-cpp/models.py`

**Review by casperdcl — DISMISSED** [review]

File: ebd-all-minilm/main.py [hunk 3]
```diff
@@ -9,6 +11,15 @@
 
 load_dotenv()
 
+MODEL_PATH = f"./ml/models/{os.getenv('MODEL_ID', 'all-MiniLM-L6-v2')}"
```
Comment (casperdcl) [inline ebd-all-minilm/main.py:14]: actually looks like this causes docker builds to fail (both for this PR and in `cht-llama-cpp`)...

```suggestion
MODEL_PATH = os.getenv('MODEL_ID', 'all-MiniLM-L6-v2')
```

wdyt @biswaroop1547?

Comment (casperdcl) [reply ebd-all-minilm/main.py:14]: also if `MODEL_PATH` is meant to be a directory (not a file), then should call it `MODEL_DIR` instead @nsosio :)

Comment (biswaroop1547) [reply ebd-all-minilm/main.py:14]: > Also the cht-llama-cpp version has  .gguf extension. Is that needed here too?

based on [this](https://github.com/premAI-io/prem-registry/pull/98/files#diff-1ae74df7e6f42efffe1c1599958b1f71c5568a9d9b55c6391380d3af317ceddeR24) I think it won't be needed.

Comment (nsosio) [reply ebd-all-minilm/main.py]: @casperdcl done :) @biswaroop1547 yes for this one we have a folder of files so no extension is needed

**Review by casperdcl — APPROVED** [review]

**Pull request merged**